src/Simulation/Simulation.jl

```diff
@@ -812,6 +812,62 @@ function refine!(sim::Simulation{T}, ::Type{WeightingPotential}, contact_id::Int
     nothing
 end
 
+"""
+    compute_min_tick_distance(grid)
+
+Compute the minimum grid spacing (tick distance) for a grid.
+Tick distances are computed as a small fraction of the physical axis lengths and then clamped to a safe numerical range.
+
+# Arguments
+- `grid` : A simulation grid (`CylindricalGrid` or `CartesianGrid3D`).
+
+# Returns
+A tuple of minimum tick distances in internal units, matching the grid type:
+- Cylindrical: `(Δr, Δφ, Δz)`
+- Cartesian: `(Δx, Δy, Δz)`
+"""
+function compute_min_tick_distance(grid::CylindricalGrid{T}) where {T}
+    min_tick = T(1e-12)
+    max_tick = T(1e-5)
+    fraction = T(1e-3)
+
+    r_axis, phi_axis, z_axis = grid.axes
+
+    r_min = T(r_axis.interval.left)
+    r_max = T(r_axis.interval.right)
+    r_len = width(r_axis.interval)
+
+    z_min = T(z_axis.interval.left)
+    z_max = T(z_axis.interval.right)
+    z_len = width(z_axis.interval)
+
+    r_mid = (r_min + r_max)/2
+    safe_r_mid = max(abs(r_mid), T(1e-30))
+
+    Δr = clamp(r_len * fraction, min_tick, max_tick)
+    Δφ = clamp(Δr / safe_r_mid, min_tick, max_tick)
+    Δz = clamp(z_len * fraction, min_tick, max_tick)
+
+    return (Δr, Δφ, Δz)
```

Comment:
We should somewhere make sure that this is type-stable and returns a `Tuple` of type `T`:

```suggestion
    Δr::T = clamp(r_len * fraction, min_tick, max_tick)
    Δφ::T = clamp(Δr / safe_r_mid, min_tick, max_tick)
    Δz::T = clamp(z_len * fraction, min_tick, max_tick)

    return (Δr, Δφ, Δz)
```